examples/102.purs

```diff
@@ -0,0 +1,34 @@
+module Main where
+
+import Prelude
+import Control.Monad.Eff (Eff)
+import Control.Monad.Eff.Class (liftEff)
+import Control.Monad.Eff.Console (CONSOLE, log)
+
+import Pipes (yield)
+import Pipes.Core ((//>), runEffect, Producer_)
+
+type Result = Int
+
+myLog
+  :: forall m
+   . Monad m
+  => String
+  -> Producer_ String m Unit
+myLog = yield
+
+foo
+  :: forall eff
+   . Int  
+  -> Int
+  -> Producer_ String (Eff eff) Result
+foo x y = do
+  myLog $ "About to compute: " <> show x <> " + " <> show y
+  r <- pure $ x + y
+  myLog $ "Computed: " <> show x <> " + " <> show y <> " = " <> show r
+  pure r
```

Comment:
- I was trying to stay as closer as possible to this http://www.haskellforall.com/2014/02/streaming-logging.html and was going to link to that one. Hence, the alias for `yield`. And also I am thinking `myLog` more explicitly communicates one possible usage of `yield`, as was done in that tutorial. But sure, I can do that for the purpose of simplifying this example. 
- Yes, effectful computation is a great idea, will do the random generation.
- And hence, rename `foo`